### DISCUSSION TITLE
Update block list

### DISCUSSION THREAD

**Discussion by eddydc** [AdguardTeam/AdGuardHome#3263] [Q&A]
Hi,
Is there a way to auto update (via cron eg) the block lists installed (active or not) for adguard?
Cheers,
Eddy

**Comment by ameshkov** [comment]
You can use curl to call the API method for checking lists updates: https://github.com/AdguardTeam/AdGuardHome/blob/master/openapi/openapi.yaml

**Reply by eddydc** [reply]
Forgive me my stupidity, but I don't see what method I should use in order to achieve this?
Do you have an example of the URL that should update the blocklist?

**Reply by ameshkov** [reply]
You need to make a POST request to `http://AGH/control/filtering/refresh`
Supply HTTP headers for basic HTTP auth.

**Reply by eddydc** [reply]
I tried it out with examples like 
curl -X POST http://AGH/control/filtering/refresh
	 -H "Content-Type: application/json" 
     -d "{\"login\":\"my_login\",\"password\":\"my_password\"}"
	 --user "login:password"   
off course filled in with the appropriate data.
I get always a forbidden.
Shorter forms like 
curl -X POST http://AGH/control/filtering/refresh
	 -H "Content-Type: application/json" 
	 --user "my_login:my_password"   
or 
curl -X POST http://AGH/control/filtering/refresh
	 --user "my_login:my_password"   
give json decode: EOF
Any suggestions?

**Reply by ameshkov** [reply]
You need to send `Authorization` header with your credentials encoded as a base64 string: https://en.wikipedia.org/wiki/Basic_access_authentication

You also need to send a JSON object like it's specified here: https://github.com/AdguardTeam/AdGuardHome/blob/master/openapi/openapi.yaml#L563

```
curl -X POST \
    -H 'Content-Type: application/json' \
    -H 'Authorization: Basic XXXXXXXX' \
    -d '{ "whitelist": false }' \
    http://AGH/control/filtering/refresh
```

**Reply by eddydc** [reply]
That worked indeed. Many thanks for the help with that.
I get a {"updated":0} back. I guess the 0 will be replaced with the actual number if anything is actually updated, no?
Looking at the whitelist param: do I need to run it twice: once for the blacklist and once for the whitelist?
